src/redis-benchmark.c

```diff
@@ -606,19 +756,31 @@ static void writeHandler(aeEventLoop *el, int fd, void *privdata, int mask) {
         c->start = ustime();
         c->latency = -1;
     }
-    if (sdslen(c->obuf) > c->written) {
+    const ssize_t buflen = sdslen(c->obuf);
+    const ssize_t writeLen = buflen-c->written;
+    if (writeLen > 0) {
         void *ptr = c->obuf+c->written;
-        ssize_t nwritten = write(c->context->fd,ptr,sdslen(c->obuf)-c->written);
-        if (nwritten == -1) {
-            if (errno != EPIPE)
-                fprintf(stderr, "Writing to socket: %s\n", strerror(errno));
-            freeClient(c);
-            return;
-        }
-        c->written += nwritten;
-        if (sdslen(c->obuf) == c->written) {
-            aeDeleteFileEvent(el,c->context->fd,AE_WRITABLE);
-            aeCreateFileEvent(el,c->context->fd,AE_READABLE,readHandler,c);
+        while(1) {
+            /* Optimistically try to write before checking if the file descriptor
+             * is actually writable. At worst we get EAGAIN. */
+            const ssize_t nwritten = writeConn(c->context,ptr,writeLen);
+            if (nwritten != writeLen) {
+                if (nwritten == -1) {
+                    if (errno != EPIPE)
+                        fprintf(stderr, "Writing to socket: %s\n", strerror(errno));
+                    freeClient(c);
+                    return;
+                }
+                if (errno != EAGAIN){
```

Comment:
isn't `EAGAIN` arrived with a `-1` return value? (in which case we already aborted before reaching this code)

Reply:
you're right. addressed in the new commit being pushed.